### ISSUE TITLE
APK

### ISSUE THREAD

**Issue by VincentGijsen** [electric-monk/pycarplay#3]
hi I was wondering i you still got the APK, or could share how/if you defeated the packager.

i'm trying to get some more details about its inner-workings and comms with the dongle, so i can (hopefully) use physical buttons to control the ui, basically like you use my own frontend to communicate with the dongle.

seems the APK i have is packaged with QiHOO, and my attemts to unpack it have failed so far. 

cheers,

**Comment by iDoka** [comment]
@VincentGijsen might I reproduce your actions on linux host to try to unpack APK?

**Comment by VincentGijsen** [comment]
hi @iDoka , I basically i used JDAX, but as the file is protected by an unpacker, you will get a very minimal sdk, and most stuff is encoded in a binary... lots of hassle,

I ended up searching for a pirated version of the dongle on the aliexpress sites... as all those vendors use their own link to a copy of the firmware on (as it seems) home-adsl connection, or something. Luckily for us those sellers of the pirated box didn't bother to add any form of protection, other than normal obfuscation by packing a APK. 

As my intention is to use the official box, but with my own interface, to combine with another project.

I found on the usual chinese shopping sites an apk by the name of 'PhoneMirrorBox'

**Comment by electric-monk** [comment]
Yes, I used JADX on macOS, but the APK that came with my dongle wasn't protected in any obvious way. I included the link that was provided with my dongle in the "download assets" shell script here: https://github.com/electric-monk/pycarplay/blob/master/downloadassets.sh#L2
